module/VuFind/src/VuFind/Session/Memcache.php

```diff
@@ -100,19 +100,17 @@ protected function connect(?Config $config, ?object $client): void
         );
 
         // Establish connection:
-        switch ($clientClass) {
-            case 'Memcache':
-                if (!$this->connection->connect($host, $port, $timeout)) {
-                    throw $connectionException;
-                }
-                break;
-            case 'Memcached':
-                $this->connection
-                    ->setOption(\Memcached::OPT_CONNECT_TIMEOUT, $timeout);
-                if (!$this->connection->addServer($host, $port)) {
-                    throw $connectionException;
-                }
-                break;
+        if ($clientClass === \Memcache::class) {
+            trigger_error('Memcache support is deprecated; use Memcached instead', E_USER_DEPRECATED);
+            if (!$this->connection->connect($host, $port, $timeout)) {
+                throw $connectionException;
+            }
+        } else {
+            $this->connection
+                ->setOption(\Memcached::OPT_CONNECT_TIMEOUT, $timeout);
```

Comment:
I'll unwrap this before merging:

```suggestion
            $this->connection->setOption(\Memcached::OPT_CONNECT_TIMEOUT, $timeout);
```